src/main/java/io/microprofile/config/ConfigProvider.java

```diff
@@ -13,28 +13,163 @@
+ * That way it is possible to overwrite configuration with lower importance from outside.</p>
+ *
+ * <p>It is also possible to register custom {@link ConfigSource}s to
+ * flexibly extend the configuration mechanism. An example would be to pick up configuration values
+ * from a database table./p>
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ *     String restUrl = ConfigProvider.getConfig().getValue("myproject.some.remote.service.url");
+ *     Integer port = ConfigProvider.getConfig().getValue("myproject.some.remote.service.port", Integer.class);
+ * </pre>
+ *
+ * </p>
  *
+ * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
+ * @author <a href="mailto:rmannibucau@apache.org">Romain Manni-Bucau</a>
+ * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
  */
-public interface ConfigProvider {
-
-	/**
-	 * Create a builder so that custom config resources can be added.
-	 * @return an instance {@link ConfigBuilder}
-	 */
-	ConfigBuilder builder();
-	/**
-	 * Get the {@link Config} after aggregating on multiple {@link ConfigSource}s
-	 * @return the {@link Config} containing all config properties
-	 */
-	Config getConfig();
+public class ConfigProvider {
+
+    private static final ConfigProviderResolver instance = ConfigProviderResolver.instance();
+
+    /**
+     * Provide a {@link Config} based on all {@link ConfigSource}s
+     * of the current Thread Context ClassLoader (TCCL)
+     *
+     * <p>There is exactly a single Config instance per ClassLoader</p>
+     */
+    public static Config getDefaultConfig() {
+        return instance.getDefaultConfig();
+    }
+
+    /**
+     * Provide a {@link Config} based on all {@link ConfigSource}s
+     * of the specified Thread Context ClassLoader (TCCL)
+     *
+     * <p>There is exactly a single Config instance per ClassLoader</p>
+     */
+    public static Config getConfig(ClassLoader cl) {
+        return instance.getConfig(cl);
+    }
+
+
+    /**
+     * Create a fresh {@link ConfigBuilder} instance.
+     * This ConfigBuilder will initially contain no
+     * {@link ConfigSource} nor any {@link Converters}.
+     * Those have to be added manually.
+     *
+     * The ConfigProvider will not manage the Config instance internally
+     */
+    public static ConfigBuilder emptyBuilder() {
+        return instance.emptyBuilder();
+    }
+    /**
+     * Create a {@link ConfigBuilder} instance.
+     * This builder contains the default {@link ConfigSource} and the {@link Converters}.
+     * @return the ConfigBuilder instance
+     */
+    public static ConfigBuilder builder() {    	
+    	return instance.builder();
+    }
+
+    /**
+     * A {@link Config} normally gets released if the ClassLoader it represents gets destroyed.
+     * Invoke this method if you like to destroy the Config prematurely.
+     */
+    public static void releaseConfig(Config config) {
+        instance.releaseConfig(config);
```

Comment:
In my book a Config is not just data. The effectively picked up ConfigSources are part of the Config. Otherwise you would not be able to pick up config changes, right?